src/car.js

```diff
@@ -0,0 +1,17 @@
+export default class Car {
+  constructor(name) {
+    this.name = name;
+    this.moveCount = 0;
+    this.result = [];
+  }
+
+  move = function (min = 0, max = 9, base = 4, actionCount) {
```

Comment:
```suggestion
  move(min = 0, max = 9, base = 4, actionCount) {
```

작성해주신 코드와 위의 코드의 차이는 무엇일까요?
프로토타입 체인 상에서 어떻게 다른지 알고 계실까요..? :smile:

Reply:
### 문법 상으로는 아래와 같은 차이가 있습니다.

#### 함수 표현식으로 정의된 인스턴스 프로퍼티(변수)
```diff
- move = function (min = 0, max = 9, base = 4, actionCount) {
``` 
#### 함수 선언식으로 정의된 클래스 프로토타입 메소드(함수)
```diff
+ move(min = 0, max = 9, base = 4, actionCount) {
``` 
---
### 클래스의 개념 상으로는 아래와 같은 차이가 있습니다.
설령 자바스크립트가 함수를 변수에 담을 수 있고 그것이 기능적으로 잘 동작하더라도,
move(차를 움직여라) 라는 기능은 차의 속성,개성이 아닌 행동(Action)이므로 메소드(함수)로 선언하는 것이 올바른 방법인 것 같습니다. 

---
### 프로토 체인 상에서의 차이는 아래와 같습니다.
#### 아래 코드는 인스턴스 프로퍼티입니다. 클래스 생성자 내부에 선언한 것과 동일합니다. 

생성자 초기화될 때 정의되며 prototype 내부에 없습니다.
prototype 내부에 없기 때문에 프로토타입 체인이 일어나지 않습니다.

```diff
- move = function (min = 0, max = 9, base = 4, actionCount) {
``` 
```
Car {name, moveCount, result}
  move : f move(min, max, base, actionCount) {
    prototype : {
      constructor : f Object()
     }
  }
  name : 
  moveCount : 
  result : 
  prototype = {
    constructor : class Car
  }
```

#### 아래 코드는 클래스 프로토타입 메소드 입니다.
Car 클래스의 prototype 객체 안에 들어갑니다.
프로토타입 체인을 이용한 상속을 제대로 활용하기 위해서는 이와 같은 프로토타입 메소드로 기능을 정의해야 합니다.

```diff
+ move(min = 0, max = 9, base = 4, actionCount) {
``` 
```
Car {name, moveCount, result}
  name : 
  moveCount : 
  result : 
  prototype = {
    constructor : class Car
    move: f move(min, max, base, actionCount)
  }
```
 
---

피드백 해주신 것 보고 내가 왜 이렇게 썼지 하고 아차 싶었습니다...
오히려 그 덕분에 정말 중요한 내용을 공부할 수 있었네요..
제가 공부하고 이해한 내용을 적어보았습니다. 조금이라도 잘못된 부분은 회초리 부탁 드립니다.

Reply:
> 제가 공부하고 이해한 내용을 적어보았습니다. 조금이라도 잘못된 부분은 회초리 부탁 드립니다.

굿입니다~  :+1: :+1: :+1: :+1:  

단순히 객체의 프로퍼티가 함수인 것과 메서드의 차이를 명확히 이해하신 것 같아서 좋습니다~